plugins/auth-backend/src/providers/oidc/provider.ts

```diff
@@ -83,39 +86,27 @@ export class OidcAuthProvider implements OAuthHandlers {
   }
 
   async refresh(req: OAuthRefreshRequest): Promise<OAuthResponse> {
-    const strategy = await this._strategy;
-    const refreshTokenResponse = await executeRefreshTokenStrategy(
-      strategy,
-      req.refreshToken,
-      req.scope,
-    );
-    const {
-      accessToken,
-      params,
-      refreshToken: updatedRefreshToken,
-    } = refreshTokenResponse;
-
-    const profile = await executeFetchUserProfileStrategy(
-      strategy,
-      accessToken,
-      params.id_token,
-    );
+    const { client } = await this.implementation;
+    const tokenset = await client.refresh(req.refreshToken);
+    if (!tokenset.access_token) {
+      throw new Error('Refresh failed');
+    }
+    const profile = await client.userinfo(tokenset.access_token);
 
     return this.populateIdentity({
       providerInfo: {
-        accessToken,
-        refreshToken: updatedRefreshToken,
-        idToken: params.id_token,
```

Comment:
this one no longer returns an id token - i presume that's intentional?

Reply:
Oh it actually should though, best add that back in

Reply:
In the OIDC implementation I was testing against, the refresh endpoint didn't return an id token.  I reasoned this as the id token wouldn't change on a refresh, only the access token.  I'll see if I can find more information one way or the other.

Reply:
Huh, typically the id token would have the same expiration as the access token, so would expect both to be refreshed

Reply:
Here's an issue describing how the provider might not return an id_token on refresh
https://github.com/panva/node-openid-client/issues/107

I'm following up with our sec team to see if there is a reason why we are not returning the id_token on refresh.  In the mean time, I'll update this PR to include the possibly undefined id_token.

Reply:
According to the OpenID Connect spec, a successful refresh response "might not contain an id_token."
https://openid.net/specs/openid-connect-core-1_0.html#RefreshTokenResponse

I've updated the PR to include it if it's present, but hopefully Backstage itself doesn't depend on it being there.

Reply:
Whether ID tokens are available or not is really gonna have to be a contract between the frontend API and the external provider implementation. If a provider is used that doesn't support ID tokens at all you wouldn't have the `getIdToken()` added to the frontend API, even though that's perhaps a bit weird for an OIDC provider 😅. If ID tokens are not included in refreshes and there's no way around that, I'd imagine that we want requests for an access tokens in the frontend go through np, but have to log in again if an ID token is required.

It's a understandable choice to not include ID tokens in refreshes as the intended usage of them is usually to just set up the initial identity and beyond that it's better to use access tokens instead. If an ID token really is needed for some reason I imagine there'd be a token endpoint where we can use our newly refreshed access token to also go fetch an ID token. It could be sane to always do that by default if possible, or make it configurable in case the consumer is typically just interested in the access token,

Either way passing it on as is (or isn't) should be fine for now. There's no dependency on the ID token from Backstage itself, or if there is it's likely a bug.